### ISSUE TITLE
Broken rendering - No handling of local GeoJSON files

### ISSUE THREAD

**Issue by okimiko** [maptiler/tileserver-gl#1324]
I'm using the latest docker release (v4.12.0) and created a custom-style based on osm-liberty, which works fine so far in vector and raster view.

Then I integrated a geojson source and added style information for its content. In vector view everything is fine and it works like expected, but in raster view there is no image (grey-screen). The preview on the index page is emtpy, too.

Here the logs, there is no http return code, no size, no duration in error case but no other error:
```
#broken
mapserver_service.1.sb87pa3vjyv0@devel-08    | GET /styles/osm-liberty-geojson/512/1/0/0.png - - - - ms
mapserver_service.1.sb87pa3vjyv0@devel-08    | GET /styles/osm-liberty-geojson/512/1/1/1.png - - - - ms
mapserver_service.1.sb87pa3vjyv0@devel-08    | GET /styles/osm-liberty-geojson/512/1/0/1.png - - - - ms
mapserver_service.1.sb87pa3vjyv0@devel-08    | GET /styles/osm-liberty-geojson/512/1/1/0.png - - - - ms

#working
mapserver_service.1.sb87pa3vjyv0@devel-08    | GET /styles/osm-liberty/512/1/0/1.png 200 214435 - 259.911 ms
mapserver_service.1.sb87pa3vjyv0@devel-08    | GET /styles/osm-liberty/512/1/0/0.png 200 372680 - 277.889 ms
mapserver_service.1.sb87pa3vjyv0@devel-08    | GET /styles/osm-liberty/512/1/1/1.png 200 211411 - 301.986 ms
mapserver_service.1.sb87pa3vjyv0@devel-08    | GET /styles/osm-liberty/512/1/1/0.png 200 399413 - 343.537 ms
```

And here the snippet of the style:
```
  "sources": {
    "openmaptiles": {
      "type": "vector",
      "url": "mbtiles://{v3}",
      "maxzoom": 7
    },
    "custom-geojson": {
      "type": "geojson",
      "data": "/files/custom.geojson"
    }
  },
[...]
```

I started a bit try and error:
- In vector view it works because my reverse proxy resolves the path
- In raster view the file does not exist because there is no relative parent
   - I tried to add a volume to my docker container at /files with fingers crossed that the file would be found on the local storage
   - It did not work :(

It did not work, too, so I started a bit code digging (serve_rendered.js):
- At the moment only special files, filetypes and http(s) is handled, so if there is no http(s), the file won't be processed
  - I think at least this should cause an error/warning and then everything else should continue as usual (without breaking the rendering)
- I tried it with an absolute url and it worked!
  - Static url seems not to be a good option for me because it is no external file

I appended the code a bit and used my previously added volume and it worked, too. Here is my change:
```
--- a/src/serve_rendered.js
+++ b/src/serve_rendered.js
@@ -1020,6 +1020,15 @@ export const serve_rendered = {
                 const format = extensionToFormat[extension] || '';
                 createEmptyResponse(format, '', callback);
               }
+            } else if (req.url[0] === '/' && fs.existsSync(req.url)) {
+              const inputFileStats = fs.statSync(req.url);
+              if (!inputFileStats.isFile() || inputFileStats.size === 0) {
+                throw Error(`File is not valid: "${req.url}"`);
+              }
+
+              fs.readFile(req.url, (err, data) => {
+                callback(err, { data: data });
+              });
             }
           },
         });
```

I'm not sure, if this really clean (at least not portable, if Windows is in the room, but then the path would not match in any case). But in general it adds an option for local geojson rendering. May be adding an additional switch for "file://" would be a good idea, too, it may help someone who is using only the rendered endpoint.

**Comment by acalcutt** [comment]
In my project at wifidb.net I use https://github.com/felt/tippecanoe to convert my geojson into mbtiles or pmtiles, then they can be served by tileserver directly. I just have a bash script like https://github.com/acalcutt/wifidb-tileserver-gl/blob/master/convert/convert.sh

It does make sense what you are saying as right now there is no handling for geojson files and no real place to put them.

I am off this week on vacation, but will look more at PRs next week

**Comment by okimiko** [comment]
Added an inital PR, but I assume, that it may make sense to implement something, that works for every endpoint.

@acalcutt: Thanks for the hint, I haven't thought over mbtiles as geojson is supported in the style spec. Have a nice vacation!

**Comment by okimiko** [comment]
Fixed in #1326 and release 4.13.2 (with some good improvements regarding parameter handling, thanks a lot) :)
